server/src/main/java/io/crate/planner/node/ddl/AlterRolePlan.java

```diff
@@ -70,13 +75,26 @@ public void executeOrFail(DependencyCarrier dependencies,
             subQueryResults
         );
         GenericProperties<Object> evaluatedProperties = alterRole.properties().map(eval);
-        Properties roleProperties = Role.Properties.of(false, evaluatedProperties);
+        if (alterRole.isReset() == false) {
+            Role.Properties.validateSessionSettings(alterRole.roleName(), evaluatedProperties, sessionSettingRegistry);
+        }
```

Comment:
I find it a bit confusing if the validation is spread across several  components (analyzer does part of it, here is another and looks like a final one in `Role.Properties.of`).

Couldn't we just have this on the lowest level in `Role.Properties.of` to ensure you cannot create a `Properties` instance with invalid settings?

Given these are rare operations, and given that inputting wrong settings is a uncommon scenario we don't need to optimize the error case with early termination.